File: visa-dev-samplecode-master/vdp-java/src/test/java/com/visa/vdp/pymntacntattrinqry/TestGeneralAttributesInquiry.java
package com.visa.vdp.pymntacntattrinqry;

import org.apache.http.client.methods.CloseableHttpResponse;
import org.testng.Assert;
import org.testng.annotations.BeforeTest;
import org.testng.annotations.Test;

import java.util.HashMap;

import org.apache.http.HttpStatus;
import com.visa.vdp.utils.VisaAPIClient;
import com.visa.vdp.utils.MethodTypes;

public class TestGeneralAttributesInquiry {

    String generalAttributeInquiry;
    VisaAPIClient visaAPIClient;

    @BeforeTest(groups = "paai")
    public void setup() {
        this.visaAPIClient = new VisaAPIClient();
        this.generalAttributeInquiry = 
                "{"
                        + "\"primaryAccountNumber\": \"4465390000029077\""
              + "}";
    }

    @Test(groups = "paai")
    public void testGeneralAttributesEnquiry() throws Exception {
        String baseUri = "paai/";
        String resourcePath = "generalattinq/v1/cardattributes/generalinquiry";

        CloseableHttpResponse response = this.visaAPIClient.doMutualAuthRequest(baseUri + resourcePath, "General Attributes Enquiry", this.generalAttributeInquiry, MethodTypes.POST, new HashMap<String, String>());
        Assert.assertEquals(response.getStatusLine().getStatusCode(), HttpStatus.SC_OK);
        response.close();
    }

}

File: vdp-python/visa/test/pymntacntattrinqry/test_general_attributes_inquiry.py
from visa.helpers.visa_api_client import VisaAPIClient
import json
import unittest
'''
@author: visa
'''

class TestGeneralAttributesEnquiry(unittest.TestCase):

    def setUp(self):
        self.visa_api_client = VisaAPIClient()
        self.general_attribute_inquiry = json.loads('''{
          "primaryAccountNumber": "4465390000029077"
        }''')
    
    def test_general_attributes_inquiry(self):
        base_uri = 'paai/'
        resource_path = 'generalattinq/v1/cardattributes/generalinquiry'
        response = self.visa_api_client.do_mutual_auth_request(base_uri + resource_path, self.general_attribute_inquiry, 'General Attributes Inquiry call', 'post')
        self.assertEqual(str(response.status_code) ,"200" ,"General attributes inquiry test failed")
        pass
